File: Civitas - Ruby/lib/sorpresa_carcel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module Civitas
  class Sorpresa_carcel < Sorpresa
    public_class_method :new
    def initialize tablero
      @tablero = tablero
    end
    
    #Aplica la carta de ir a la carcel
    def aplicar_a_jugador(actual, todos)
      if(jugador_correcto(actual, todos))
        informe(actual, todos)
        todos[actual].encarcelar(@tablero.num_casilla_carcel)
      end
    end
    
  end
end

File: Civitas - Java/src/civitas/SorpresaCarcel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author juanc
 */
public class SorpresaCarcel extends Sorpresa {
    
    //Para ir a la carcel
    SorpresaCarcel (Tablero tablero){
        super(tablero);
    }
    
    //Aplica la carta de ir a la carcel
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos){
        
        if(jugadorCorrecto(actual, todos) ){
            System.out.println("SE VA APLICAR UNA SORPRESA DE TIPO CARCEL");
            informe(actual, todos);
            todos.get(actual).encarcelar(tablero.getCarcel());
        }
        
    }
    
}
